config/src/parameters/actual.rs

```diff
@@ -100,27 +100,20 @@ pub enum Genesis {
     },
     /// The peer is responsible for submitting the genesis block
     Full {
-        /// Genesis account key pair
-        key_pair: KeyPair,
-        /// Path to the [`RawGenesisBlock`]
+        /// Genesis account public key
+        public_key: PublicKey,
+        /// Path to the genesis file
         file: PathBuf,
+        /// Hex-encoded genesis config
+        signature: String,
```

Comment:
Consider using a stronger type right away. Possibly a newtype around `iroha_genesis::GenesisSignature` which will implement `Deserialize` and `FromStr` from a hex string.